L1Trigger/L1THGCalUtilities/python/clustering3d.py

```diff
@@ -1,106 +1,105 @@
-    producer.ProcessorParameters.C3d_parameters.dR_multicluster_byLayer = cms.vdouble(distances)
+                               distances=histoMaxVariableDR_C3d_params.dR_multicluster_byLayer_coefficientA,
+                               nBins_R=histoMaxVariableDR_C3d_params.nBins_R_histo_multicluster,
+                               nBins_Phi=histoMaxVariableDR_C3d_params.nBins_Phi_histo_multicluster,
+                               binSumsHisto=histoMaxVariableDR_C3d_params.binSumsHisto,
+                               seed_threshold=histoMaxVariableDR_C3d_params.threshold_histo_multicluster,
+                               ):
+    producer = process.hgcalBackEndLayer2Producer.clone()
+    producer.ProcessorParameters.C3d_parameters = histoMax_C3d_params.clone()
+    set_histomax_params(producer.ProcessorParameters.C3d_parameters, 0, nBins_R, nBins_Phi, binSumsHisto, seed_threshold)
+    producer.ProcessorParameters.C3d_parameters.dR_multicluster_byLayer_coefficientA = distances
+    producer.InputCluster = cms.InputTag('{}:HGCalBackendLayer1Processor2DClustering'.format(inputs))
     return producer
 
 
-def create_histoInterpolatedMax(process, inputs,
-        distance = 0.03,
-        nBins_R = 36,
-        nBins_Phi = 216,
-        binSumsHisto = binSums,
-        ):
-    producer = create_histoMax( process, inputs, distance, nBins_R, nBins_Phi, binSumsHisto )
-    producer.ProcessorParameters.C3d_parameters.type_multicluster = cms.string('HistoInterpolatedMaxC3d')
-    return producer
-
-def create_histoInterpolatedMax1stOrder(process, inputs):
-    producer = create_histoInterpolatedMax( process, inputs )
-    producer.ProcessorParameters.C3d_parameters.neighbour_weights=cms.vdouble(  0    , 0.25, 0   ,
-                                                   0.25 , 0   , 0.25,
-                                                   0    , 0.25, 0
-                                                )
+def create_histoInterpolatedMax1stOrder(process, inputs,
+                                        distance=histoInterpolatedMax_C3d_params.dR_multicluster,
+                                        nBins_R=histoInterpolatedMax_C3d_params.nBins_R_histo_multicluster,
+                                        nBins_Phi=histoInterpolatedMax_C3d_params.nBins_Phi_histo_multicluster,
+                                        binSumsHisto=histoInterpolatedMax_C3d_params.binSumsHisto,
+                                        seed_threshold=histoInterpolatedMax_C3d_params.threshold_histo_multicluster,
+                                        ):
+    producer = process.hgcalBackEndLayer2Producer.clone()
+    producer.ProcessorParameters.C3d_parameters = histoInterpolatedMax_C3d_params.clone()
+    set_histomax_params(producer.ProcessorParameters.C3d_parameters, distance, nBins_R, nBins_Phi, binSumsHisto, seed_threshold)
+    producer.InputCluster = cms.InputTag('{}:HGCalBackendLayer1Processor2DClustering'.format(inputs))
```

Comment:
Probably better to add the modification of the kernel in case the default value changes in the future, similarly to the 2ndOrder version:
```python
producer.ProcessorParameters.C3d_parameters.neighbour_weights = neighbour_weights_1stOrder
```